### ISSUE TITLE
libopenshot isn't compatible with ImageMagick 7

### ISSUE THREAD

**Issue by ferdnyc** [OpenShot/libopenshot#168]
In #162 @cyberhoffman attempted to build libopenshot-0.2.0 with ImageMagick 7, and was unsuccessful. A [build log](https://paste.pound-python.org/show/3mFWhQebpMzZuWWnqM1f/) was helpfully supplied, which indicated that the first compilation error occurs here:
```
/var/tmp/portage/media-libs/libopenshot-0.2.0/work/libopenshot-0.2.0/src/Frame.cpp: 
В функции-члене «std::shared_ptr<Magick::Image> openshot::Frame::GetMagickImage()»:
/var/tmp/portage/media-libs/libopenshot-0.2.0/work/libopenshot-0.2.0/src/Frame.cpp:905:16: 
ошибка: «using element_type = class Magick::Image {aka class Magick::Image}» has no 
member named «matte»
  magick_image->matte(true);
                ^~~~~
```

That's ImageMagick/ImageMagick#236, which details that the `Magick::Image` class's `matte()` attribute/method has been renamed to `alpha()`. 

Since `alpha()` is not part of IM6, the change makes the two APIs incompatible with no common ground, necessitating conditionalization of the code in order to support building with either version. For this purpose, the `MagickLibVersion` macro is `#define`'d in the headers for both IM6 and IM7, its value is a hexadecimal integer representation of the ImageMagick version (`0x699` for ImageMagick 6.9.9.x, `0x708` for ImageMagick 7.0.8.x, etc.)

This is merely the first issue encountered, there may be additional incompatibilities in the libopenshot code. Until it successfully builds all the way through with ImageMagick-7 we won't know for sure.

**Comment by jeffski** [comment]
Would love to see support for ImageMagick7. I'm keen to make some enhancements to the TextReader, one of which is text alignment but looks like what I'd want is https://www.imagemagick.org/api/Magick++/classMagick_1_1DrawableTextAlignment.html which is only available in V7.

**Comment by steils** [comment]
Probably duplicate of https://github.com/OpenShot/libopenshot/issues/60

**Comment by ferdnyc** [comment]
I'm actually working on this right now. Should have a PR tomorrow. (You can thank [Remi Collet](https://blog.remirepo.net/), indirectly, who did the work of finally getting ImageMagick7 RPMs together for Fedora.)

**Comment by ferdnyc** [comment]
(Doesn't help that frickin' imagemagick.org has been **down** since at least last night, just when I'm trying to access reference materials!) :angry:

**Comment by steils** [comment]
@ferdnyc https://github.com/OpenShot/libopenshot/pull/250

**Comment by ferdnyc** [comment]
Ah, cool! :+1:  I actually went a different route, with a `MagickUtilities.h` (analogous to the existing `FFmpegUtilities.h` and `OpenMPUtilities.h`) to do most of the compatibility lifting. I guess we'll play dueling PRs! ...The code looks practically the same, effectively anyway.

**Comment by ferdnyc** [comment]
Closing as fixed by the merge of #252
